### PR TITLE
Add #include<complex.h> to the include list

### PR DESCRIPTION
**Pull request by IgorBaratta**
Insert `#include<complex.h>` when the scalar type is defined as `double complex`.
This allows dolfinx to use the generated c code (without throwing an error), even though it it does not use complex scalars yet.

Alternatively, this could be done by the backend (more changes needed).

### CODE REVIEW

File: ffc/formatting.py [hunk 1]
```diff
@@ -177,6 +178,11 @@ def _generate_includes(includes, parameters):
...
     # s2 = external_includes - s
+    includes_h = "\n".join(sorted(s)) + "\n" if s else ""
 
-    includes = "\n".join(sorted(s)) + "\n" if s else ""
-    return includes
+    # This should really be set by the backend
+    if "scalar_type" in parameters.keys():
+        if parameters["scalar_type"] == "double complex":
+            includes_c = "#include<complex.h>" + "\n"
+
```
Comment (garth-wells) [inline ffc/formatting.py]: Maybe a bit more like:
```
scalar_type = parameters.get("scalar_type")
if scalar_type is "scalar_type": 
    includes_c = "#include<complex.h>\n"
```

File: ffc/formatting.py [hunk 2]
```diff
@@ -177,6 +178,11 @@ def _generate_includes(includes, parameters):
     s = set(default_h_includes) | includes
 
     # s2 = external_includes - s
+    includes_h = "\n".join(sorted(s)) + "\n" if s else ""
```
Comment (garth-wells) [inline ffc/formatting.py]: Put on two lines.

File: ffc/formatting.py (hunk 1 above)
Comment (chrisrichardson) [reply ffc/formatting.py]: you mean `if scalar_type is "double complex":`

**Comment by garth-wells** [comment]
Is there a reason why `complex.h` isn't included in the header file? I would have thought the complex type would appear in the interface (function signatures).

File: ffc/formatting.py (hunk 1 above)
Comment (IgorBaratta) [reply ffc/formatting.py]: Much better.
I'll keep `==` instead of `is`, otherwise the if would never be true.

**Comment by IgorBaratta** [comment]
I was including `complex.h`  in header file, however the error persisted.
Maybe it's because the the complex type signature is not defined in the header file.
I'll look into it.

**Comment by IgorBaratta** [comment]
I was including `complex.h` in the header file, however the error persisted.
Maybe it's because the the complex type signature is not defined in the header file.
I'll look into it.

**Comment by IgorBaratta** [comment]
I'll work further on the complex case, including the signature of tabulate_tensor. And then open a new PR.

**Pull request closed without merging**